public/app/features/alerting/unified/Silences.test.tsx

```diff
@@ -0,0 +1,260 @@
+
+const mocks = {
+  api: {
+    fetchSilences: typeAsJestMock(fetchSilences),
+    fetchAlerts: typeAsJestMock(fetchAlerts),
+    createOrUpdateSilence: typeAsJestMock(createOrUpdateSilence),
+  },
+};
+
+const renderSilences = (location = '/alerting/silences/') => {
+  const store = configureStore();
+  locationService.push(location);
+
+  return render(
+    <Provider store={store}>
+      <Router history={locationService.getHistory()}>
+        <Silences />
+      </Router>
+    </Provider>
+  );
+};
+
+const dataSources = {
+  am: mockDataSource({
+    name: 'Alertmanager',
+    type: DataSourceType.Alertmanager,
+  }),
+};
+
+const ui = {
+  silencesTable: byTestId('silences-table'),
+  silenceRow: byTestId('silence-table-row'),
+  silencedAlertCell: byTestId('silenced-alerts'),
+  queryBar: byPlaceholderText('Search'),
+  editor: {
+    timeRange: byLabelText('Timepicker', { exact: false }),
+    timeRangeStart: byRole('textbox', { name: /timepicker from field/i }),
+    timeRangeEnd: byRole('textbox', { name: /timepicker to field/i }),
+    applyTimeRange: byText(/apply time range/i),
+    durationField: byLabelText('Duration'),
+    durationInput: byRole('textbox', { name: /duration/i }),
+    matchersField: byTestId('matcher'),
+    matcherName: byPlaceholderText('label'),
+    matcherValue: byPlaceholderText('value'),
+    comment: byPlaceholderText('Details about the silence'),
+    createdBy: byPlaceholderText('Username'),
+    matcherOperatorSelect: byLabelText('operator'),
+    matcherOperator: (operator: MatcherOperator) => byText(operator, { exact: true }),
+    addMatcherButton: byRole('button', { name: 'Add matcher' }),
+    submit: byText('Submit'),
+  },
+};
+
+const resetMocks = () => {
+  mocks.api.fetchSilences.mockImplementation(() => {
+    return Promise.resolve([
+      mockSilence({ id: '12345' }),
+      mockSilence({ id: '67890', matchers: parseMatchers('foo!=bar'), comment: 'Catch all' }),
+    ]);
+  });
+
+  mocks.api.fetchAlerts.mockImplementation(() => {
+    return Promise.resolve([
+      mockAlertmanagerAlert({
+        labels: { foo: 'bar' },
+        status: { state: AlertState.Suppressed, silencedBy: ['12345'], inhibitedBy: [] },
+      }),
+      mockAlertmanagerAlert({
+        labels: { foo: 'buzz' },
+        status: { state: AlertState.Suppressed, silencedBy: ['67890'], inhibitedBy: [] },
+      }),
+    ]);
+  });
+
+  mocks.api.createOrUpdateSilence.mockResolvedValue(mockSilence());
+};
+
+describe('Silences', () => {
+  beforeAll(resetMocks);
+  afterEach(resetMocks);
+
+  beforeEach(() => {
+    setDataSourceSrv(new MockDataSourceSrv(dataSources));
+  });
+
+  it('loads and shows silences', async () => {
+    renderSilences();
+    await waitFor(() => expect(mocks.api.fetchSilences).toHaveBeenCalled());
+    await waitFor(() => expect(mocks.api.fetchAlerts).toHaveBeenCalled());
+
+    expect(ui.silencesTable.query()).not.toBeNull();
+    expect(ui.silenceRow.queryAll()).toHaveLength(2);
```

Comment:
it would be great to check if key info is rendered as well, not only number of rows